lms/templates/courseware/course_about.html

```diff
@@ -310,7 +310,7 @@
 
           ##<li class="important-dates-item"><i class="icon fa fa-clock-o"></i><p class="important-dates-item-title">${_('Course Length')}</p><span class="important-dates-item-text course-length">${_('{number} weeks').format(number=15)}</span></li>
 
-          %if course_price and (is_shopping_cart_enabled or is_cosmetic_price_enabled):
+          %if course_price and (is_cosmetic_price_enabled or (is_shopping_cart_enabled and registration_price)):
```

Comment:
Instead of passing `registration_price` through to the template, we could pass a variable that's like, `show_price` (or some variable name that's very explicit and clear), and define that in views.py to be `show_price = is_shopping_cart_enabled and registration_price`. That might clear up the logic in this template a bit more.